src/m2khardwaretrigger_v0.24_impl.cpp

```diff
@@ -187,9 +199,13 @@ M2K_TRIGGER_SOURCE_DIGITAL M2kHardwareTriggerV024Impl::getDigitalSource() const
 
 M2K_TRIGGER_SOURCE_DIGITAL M2kHardwareTrigger::getDigitalSource() const
 {
-	throw std::runtime_error("M2kHardwareTrigger: "
-				 "the digital external source is not available on "
-				 "the current board; Check the firmware version.");
+	throw m2k_exception::make("M2kHardwareTrigger: "
```

Comment:
Why not use the throw macro? Will it be removed?

Reply:
The macro uses throw_exception function, which will throw the exception only if exceptions are enabled. In these functions (from m2khardwaretrigger) the usage of the macro will cause a compile error, because when the exceptions are disabled the functions won't return anything.

Reply:
So with exceptions disabled this hardware trigger "things" will still throw exceptions? Wouldn't it be more consistent to throw an exception when exceptions are enabled and return `nullptr` when exceptions are disabled?

Reply:
Modified.